src/RlnBase.sol

```diff
@@ -278,4 +281,15 @@ abstract contract RlnBase {
         }
         return commitments;
     }
+
+    function root() public view returns (uint256) {
```

Comment:
related to https://github.com/vacp2p/rln-contract/pull/36/files#r1500330402

does this work with hundred or thousand of leafs? `calcRoot` has to do tons of hashes.

> even view functions have limits so I suspect with a high amount of leaves.length this wont work, since too many hashes to do.

Reply:
500M seems too high.

> https://community.infura.io/t/gas-limit-on-eth-call/1115
Says: 19 981 536

https://docs.infura.io/api/networks/ethereum/json-rpc-methods/eth_estimategas
> To prevent abuse of the API, the gas parameter in this eth_estimateGas method and in [eth_call](https://docs.infura.io/api/networks/ethereum/json-rpc-methods/eth_call) is capped at 10x (1000%) the current block gas limit. You can recreate this behavior in your local test environment (besu, geth, or other client) via the [rpc.gascap](https://geth.ethereum.org/docs/fundamentals/command-line-options#commands) command-line option.

and from geth 50 000 000
> From geth
--rpc.gascap value                  (default: 50000000)


How many leafs do we need to hit this limit?

Reply:
https://docs.alchemy.com/reference/eth-estimategas#:~:text=NOTE%3A%20this%20parameter%20has%20a,used%20for%20each%20paid%20gas.

Reply:
caps out at 255 leaves, not ideal, but perhaps we can split the computation and make the calcRoot function accept an array of prefilled nodes of the tree, and make multiple rpc calls to calcRoot like pagination

Reply:
ow yeap quite small tree which makes it not practical. It could be interesting to do so, but perhaps with that it loses the purpose. I mean, with this a client has to keep calling back and forth, downloading the leafs, keeping some local state, which is error prone and makes me wonder if its practical. Of course its more gas efficient than the LazyIMT approach, so we should asses what matters more.

Reply:
applications which will have smaller trees can probably use this - and have the benefits that this approach brings. will think of another approach how we can consume less gas and make the least amount of rpc calls.